yomigeek/mtracker-app
# #157754355 User Modify Request API

#### What does this PR do?
* Create an API endpoint to enable a user modify a pending user request

#### Description of Task to be completed?
* User should be able to update a request through this API endpoint

#### How should this be manually tested?
* Clone the project to your local repository from git clone https://github.com/yomigeek/mtracker-app.git
* All necessary dependencies as stated in the README.md file should be installed
* run $ npm install
* Run app locally using $ npm run start
* Open the postman application and enter the URL http://localhost:5000/users/requests/:requestId with PUT request method selected

* After the above is successful, you should get the following response: 
```
{
    "status": "success",
    "message": "Request updated successfully!",
    "data": {
        "id": 2,
        "title": "My fridge door is poor",
        "description": "Fridge door is faulty and doing bad greatly",
        "priority": "medium"
    }
}
```
#### Any background context you want to provide?
To get the above response a request must have been created otherwise a response of no request exist, the below response will be displayed instead

```
{
    "status": 'fail',
    "message": 'Invalid Request Id',
}
```

####  What are the relevant pivotal tracker stories?
story type: feature
story id: 157754355

#### What are the relevant Github Issues?
None

#### Questions:
None

![modify_admin](https://user-images.githubusercontent.com/5827585/40688884-bbd88d58-6397-11e8-98d8-bce2bd209a59.PNG)
